lib/liveman/user/users.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Liveman.User.Users do
+  @email_regex_format ~r/^\S+@\S+\.\S+$/
+
+  def validate_registration_params(params) do
```

Comment:
When using validation, this should be done instead at the schema level. Either use the default 	`changeset` method or create another one e.g. `registration_changeset`.

Then in the context, there can be a method called `create_user` or `register_user`:

```
def register_user(user_params) do
  User |> User.changeset(user_params) |> Repo.insert()
end
```

This will raised a changeset error when validation fails.

Then in the controller, the standard way would be:

```
case Users.register_user(params) do
  {:ok, user} ->
    json(conn, %{meta: %{detail: "An confirmation email has been sent with an OTP code"}})

  {:error, %Ecto.Changeset{} = changeset} ->
     conn
     |> put_status(:unprocessable_entity)
     |> render(ErrorView, "error.json", errors: [%{detail: changeset.errors}])
end
```

Reply:
Thanks @olivierobert , I changed it accordingly